crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -628,6 +630,30 @@ impl TryFrom<i32> for ManifestContentType {
     }
 }
 
+impl ManifestListEntry {
```

Comment:
I'm confused by the naming, should this be a `ManifestFile`?

From the [spec](https://iceberg.apache.org/spec/#manifest-lists): 

Manifest list files store `manifest_file`, a struct with the following fields

Reply:
Yes, I think we will need to some rename to make it more similar to java/python implementation.

Reply:
Tracked in #145